pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -542,21 +542,25 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 				status.UpdatedReplicas++
 			}
 			// if the set does not allow bursting, return immediately
-			if monotonic || decreaseAndCheckMaxUnavailable(scaleMaxUnavailable) {
+			if monotonic {
 				return &status, nil
+			} else if decreaseAndCheckMaxUnavailable(scaleMaxUnavailable) {
+				break
```

Comment:
klog.V(4).Infof(
					"StatefulSet %s/%s Pod %s is Creating, and break pods scale",
					set.Namespace,
					set.Name,
					replicas[i].Name)